### PR TITLE
draft: Update orchestration spec

### PR DESCRIPTION
**Pull request by jjtang1985**
This PR should not be merged.
Let's re-generate the client and compare with the current client.

### CODE REVIEW

File: packages/orchestration/src/client/api/schema/orchestration-config.ts [hunk 1]
```diff
@@ -4,9 +4,15 @@
...
+import type { GlobalStreamOptions } from './global-stream-options.js';
 /**
  * Representation of the 'OrchestrationConfig' schema.
  */
 export type OrchestrationConfig = {
   module_configurations: ModuleConfigs;
+  /**
+   * If true, the response will be streamed back to the client
+   */
+  stream?: boolean;
```
Comment (jjtang1985) [inline packages/orchestration/src/client/api/schema/orchestration-config.ts:16]: streaming

File: packages/orchestration/src/client/api/schema/output-filtering-config.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ * Copyright (c) 2024 SAP SE or an SAP affiliate company. All rights reserved.
+ *
+ * This is a generated file powered by the SAP Cloud SDK for JavaScript.
+ */
+import type { FilterConfig } from './filter-config.js';
+import type { FilteringStreamOptions } from './filtering-stream-options.js';
+/**
+ * Representation of the 'OutputFilteringConfig' schema.
+ */
+export type OutputFilteringConfig = {
```
Comment (jjtang1985) [inline packages/orchestration/src/client/api/schema/output-filtering-config.ts:11]: breaking change, together with InputFilteringConfig

File: packages/orchestration/src/client/api/schema/template.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ *       "role": "user",
+ *       "content": "How can the features of AI in SAP BTP specifially {​{?product}}, be applied to {​{?inputContext}}"
+ *     }
+ *   ],
+ *   "defaults": {
+ *     "inputContext": "The default text that will be used in the template if inputContext is not set"
+ *   }
+ * }
+ */
+export type Template = {
```
Comment (jjtang1985) [inline packages/orchestration/src/client/api/schema/template.ts:21]: breaking change?

File: packages/orchestration/src/client/api/schema/templating-module-config.ts [hunk 4]
```diff
@@ -3,28 +3,8 @@
...
-  /**
-   * A chat message array to be formatted with values from input_params. Both role and content can be templated. If messages_history is provided, the templated messages will be appended.
-   */
-  template: ChatMessages;
-  /**
-   * Optional default values for the template. If a parameter has no default it is required.
-   */
-  defaults?: Record<string, any>;
-} & Record<string, any>;
+export type TemplatingModuleConfig = Template;
```
Comment (jjtang1985) [inline packages/orchestration/src/client/api/schema/templating-module-config.ts:10]: breaking change?

File: packages/orchestration/src/client/api/schema/azure-content-safety.ts [hunk 5]
```diff
@@ -5,11 +5,11 @@
  */
 import type { AzureThreshold } from './azure-threshold.js';
 /**
- * Filter configuration for Azure Content Safety.
+ * Filter configuration for Azure Content Safety
  */
-export interface AzureContentSafety {
+export type AzureContentSafety = {
```
[the remark this answers is not in the document]
Comment (jjtang1985) [reply packages/orchestration/src/client/api/schema/azure-content-safety.ts:10]: Due to eslint thing.

**Comment by jjtang1985** [comment]
@deekshas8, please let me know, if we can close this PR.

**Comment by deekshas8** [comment]
Will be handled in the streaming ticket

**Pull request closed without merging**